### PR TITLE
feat: add notifications on footer on mobile

### PR DESCRIPTION
**Pull request by omBratteng**
# Changes

Moves notifications to bottom

image

AS-398

### Preview domain
https://as-398-notification-micro.preview.app.daily.dev

### CODE REVIEW

File: packages/webapp/components/footer/FooterNavBarTabs.tsx [hunk 1]
```diff
@@ -82,12 +88,45 @@ const Tab = ({ tab, isActive }: TabProps) => {
...
+      {!!unreadCount && (
+        <Bubble
+          className={classNames('-right-1.5 -top-1.5 cursor-pointer px-1')}
+        >
+          {getUnreadText(unreadCount)}
+        </Bubble>
+      )}
+    </div>
+  );
+};
```
Comment (omBratteng) [inline packages/webapp/components/footer/FooterNavBarTabs.tsx:106]: I wanted to re-use the `NotificationsBell` component, but it adds a tooltip and an extra button that doesn't fit in the bottom navbar. So for experiment sake, I thought of just doing a local component here.

File: packages/webapp/components/footer/FooterNavBarTabs.tsx [hunk 2]
```diff
@@ -82,12 +88,45 @@ const Tab = ({ tab, isActive }: TabProps) => {
...
+      mobileUxTabs.splice(-1, 0, {
+        requiresLogin: true,
+        path: '/notifications',
+        title: 'Notifications',
+        icon: (active: boolean) => <Notifications active={active} />,
+      });
+    }
+
+    return mobileUxTabs;
+  }, [notificationsNavBar]);
```
Comment (omBratteng) [inline packages/webapp/components/footer/FooterNavBarTabs.tsx:125]: Had to memoize it, unless you'd end up with lots of notification buttons
image

File: packages/shared/src/components/profile/Header.tsx [hunk 3]
```diff
@@ -30,30 +34,37 @@ export function Header({
     cid: ReferralCampaignKey.ShareProfile,
     logObject: () => ({ event_name: 'share profile', target_id: user.id }),
   });
+  const isMobile = useViewSize(ViewSize.MobileL);
+  const notificationsNavBar = useFeature(feature.notificationsNavBar);
```
Comment (rebelchris) [inline packages/shared/src/components/profile/Header.tsx:38]: We probably want to limit enrollment to only mobile users, maybe use `useConditionalFeature` here to be safe :)

File: packages/shared/src/hooks/useActiveNav.tsx [hunk 4]
```diff
@@ -16,6 +18,7 @@ export interface UseActiveNav {
 export default function useActiveNav(activeFeed: AllFeedPages): UseActiveNav {
   const router = useRouter();
   const isLaptop = useViewSize(ViewSize.Laptop);
+  const notificationsNavBar = useFeature(feature.notificationsNavBar);
```
Comment (rebelchris) [inline packages/shared/src/hooks/useActiveNav.tsx:21]: Here we probably also want to hard enroll only to mobile?
(also note the current one is `!laptop` which might be wrong as we want hard `isMobile`)

File: packages/webapp/components/footer/FooterNavBarTabs.tsx [hunk 5]
```diff
@@ -82,12 +88,45 @@ const Tab = ({ tab, isActive }: TabProps) => {
...
+        </Bubble>
+      )}
+    </div>
+  );
+};
+
 export function FooterNavBarTabs({
   activeTab,
 }: FooterNavBarContainerProps): ReactElement {
+  const notificationsNavBar = useFeature(feature.notificationsNavBar);
```
Comment (rebelchris) [inline packages/webapp/components/footer/FooterNavBarTabs.tsx:111]: Can you check if this component only get's rendered on mobile?
Else we'll need to do enrollment check here as well.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Looks good, just some enrollment check we need to add.

File: packages/webapp/components/footer/FooterNavBarTabs.tsx (hunk 5 above)
Comment (omBratteng) [reply packages/webapp/components/footer/FooterNavBarTabs.tsx:111]: FooterNavBarTabs is rendered by FooterNavBar if `showNav` is set to true, which is set to true in FooterNavBarLayout if isMobile.

So, only on mobile.

File: packages/shared/src/components/profile/Header.tsx (hunk 3 above)
Comment (omBratteng) [reply packages/shared/src/components/profile/Header.tsx]: Used useConditionalFeature

File: packages/shared/src/hooks/useActiveNav.tsx (hunk 4 above)
Comment (omBratteng) [reply packages/shared/src/hooks/useActiveNav.tsx]: Added useConditionalFeature and an extra check for !isMobile where it appends the notifications to array.

**Review by nensidosari — APPROVED** [review]
Looks good

**Review by rebelchris — APPROVED** [review]

**Pull request merged**